### ISSUE TITLE
ActionView::Template::Error (No route matches 

### ISSUE THREAD

**Issue by cdmo** [psu-libraries/psulib_blacklight#362]
```
[2019-06-18T14:00:55.977879 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Started GET "/catalog/1128228" for 128.118.152.108 at 2019-06-18 14:00:55 -0400
I, [2019-06-18T14:00:55.980101 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Processing by CatalogController#show as HTML
I, [2019-06-18T14:00:55.980245 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Parameters: {"id"=>"1128228"}
D, [2019-06-18T14:00:55.990413 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Solr query: get get {:qt=>nil, :ids=>"1128228"}
D, [2019-06-18T14:00:55.990549 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Solr fetch (5.1ms)
D, [2019-06-18T14:00:55.993981 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Search Load (0.6ms)  SELECT  `searches`.* FROM `searches` WHERE `searches`.`id` IN (3744, 3743, 3741, 3740, 3739, 3738, 3718, 3666, 3650, 3649, 3612, 3611, 3610, 3586, 3585, 3579) AND `searches`.`id` = 3744 ORDER BY updated_at desc LIMIT 1
D, [2019-06-18T14:00:56.717115 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Solr query: get select {"qt"=>nil, "facet.field"=>["access_facet", "format", "{!ex=campus_facet_single}campus_facet", "{!ex=library_facet_single}library_facet", "{!ex=up_library_facet_single}up_library_facet", "pub_date_itsi", "language_facet", "subject_topic_facet", "genre_facet", "media_type_facet", "lc_1letter_facet", "lc_rest_facet"], "facet.query"=>[], "facet.pivot"=>["lc_1letter_facet,lc_rest_facet"], "fq"=>[], "hl.fl"=>[], "rows"=>2, "facet"=>false, "f.campus_facet.facet.sort"=>"index", "f.campus_facet.facet.limit"=>-1, "f.library_facet.facet.sort"=>"index", "f.library_facet.facet.limit"=>-1, "f.up_library_facet.facet.sort"=>"index", "f.up_library_facet.facet.limit"=>-1, "f.language_facet.facet.limit"=>11, "f.subject_topic_facet.facet.limit"=>21, "f.genre_facet.facet.limit"=>21, "f.media_type_facet.facet.limit"=>21, "f.lc_1letter_facet.facet.sort"=>"index", "f.lc_rest_facet.facet.sort"=>"index", "sort"=>"score desc, pub_date_itsi desc, title_sort asc", "stats"=>"true", "stats.field"=>["pub_date_itsi"], "fl"=>"id"}
D, [2019-06-18T14:00:56.717250 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Solr fetch (722.4ms)
I, [2019-06-18T14:00:56.718171 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendering catalog/show.html.erb within layouts/blacklight
I, [2019-06-18T14:00:56.721927 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered catalog/_show_header_default.html.erb (0.3ms)
I, [2019-06-18T14:00:56.726045 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered catalog/_show_top_fields.html.erb (3.9ms)
I, [2019-06-18T14:00:56.726361 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered catalog/_show_availability.html.erb (0.1ms)
D, [2019-06-18T14:00:56.726796 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Looking for document partial show_list_book
D, [2019-06-18T14:00:56.727116 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Looking for document partial show_list_default
D, [2019-06-18T14:00:56.727380 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Looking for document partial show_book
D, [2019-06-18T14:00:56.727560 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Looking for document partial show_default
I, [2019-06-18T14:00:56.736123 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered catalog/_marc_record_details.html.erb (0.3ms)
I, [2019-06-18T14:00:56.746183 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered catalog/_show_main_content.html.erb (27.7ms)
I, [2019-06-18T14:00:56.746395 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered catalog/show.html.erb within layouts/blacklight (28.1ms)
I, [2019-06-18T14:00:56.746719 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendering layouts/blacklight/base.html.erb
I, [2019-06-18T14:00:56.748113 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered shared/_announcement.html.erb (0.2ms)
D, [2019-06-18T14:00:56.750041 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   User Load (0.5ms)  SELECT  `users`.* FROM `users` WHERE `users`.`email` = 'guest_zscWBRGtbwR72ByLDAMQ_1560188736_1@example.com' LIMIT 1
D, [2019-06-18T14:00:56.752121 #2666] DEBUG -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]    (0.3ms)  SELECT COUNT(*) FROM `bookmarks` WHERE `bookmarks`.`user_id` = 771 AND `bookmarks`.`user_type` = 'User'
I, [2019-06-18T14:00:56.752330 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered vendor/bundle/ruby/2.5.0/bundler/gems/blacklight-39d676481858/app/views/blacklight/nav/_bookmark.html.erb (1.7ms)
I, [2019-06-18T14:00:56.752748 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered shared/_user_util_links.html.erb (4.0ms)
I, [2019-06-18T14:00:56.757085 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered catalog/_search_form.html.erb (3.9ms)
I, [2019-06-18T14:00:56.757209 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered shared/_header_navbar.html.erb (8.9ms)
I, [2019-06-18T14:00:56.757543 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered vendor/bundle/ruby/2.5.0/bundler/gems/blacklight-39d676481858/app/views/shared/_flash_msg.html.erb (0.1ms)
I, [2019-06-18T14:00:56.758724 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]   Rendered layouts/blacklight/base.html.erb (11.9ms)
I, [2019-06-18T14:00:56.758973 #2666]  INFO -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] Completed 500 Internal Server Error in 779ms (ActiveRecord: 1.4ms)
F, [2019-06-18T14:00:56.759827 #2666] FATAL -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]
F, [2019-06-18T14:00:56.759897 #2666] FATAL -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] ActionView::Template::Error (No route matches {:action=>"index", :controller=>"advanced", :id=>"27043523", :page=>1}):
F, [2019-06-18T14:00:56.760069 #2666] FATAL -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]     30:         <% if @search_context %>
[0333aef5-b671-4743-b3ba-ae5fadd0cfb4]     31:             <% if current_search_session %>
[0333aef5-b671-4743-b3ba-ae5fadd0cfb4]     32:                 <div id="appliedParams" class="col-auto">
[0333aef5-b671-4743-b3ba-ae5fadd0cfb4]     33:                   <%= link_back_to_catalog class: 'btn btn-outline-secondary btn-sm' %>
[0333aef5-b671-4743-b3ba-ae5fadd0cfb4]     34:                 </div>
[0333aef5-b671-4743-b3ba-ae5fadd0cfb4]     35:             <% end %>
[0333aef5-b671-4743-b3ba-ae5fadd0cfb4]     36:
F, [2019-06-18T14:00:56.760115 #2666] FATAL -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4]
F, [2019-06-18T14:00:56.760157 #2666] FATAL -- : [0333aef5-b671-4743-b3ba-ae5fadd0cfb4] app/views/layouts/blacklight/base.html.erb:33:in `_app_views_layouts_blacklight_base_html_erb__2452668260066727727_69883618903300'
```

**Comment by cdmo** [comment]
Think I got it. To reproduce:

1. Do a search
2. Click an item
3. Click advanced search link
4. Without performing another search, go to another record directly. For example, paste in the URL that resulted from step 2.
5. Error

Here's what's happening:

The Blacklight Advanced Search gem has a generator that inserts this link into views:

`blacklight_advanced_search_engine.advanced_search_path(search_state.to_h)`

That last piece `search_state.to_h` is the culprit I think. For a deeper reason, the `search_state.to_h` after click on an item view becomes `id:NNNNNN`. When you go to Advanced Search that gets past along. Now, if you don't perform another search that wipes out your sessions's query parameters, when you go to a record view it somehow wants to pull up the `index` view for the `advanced` controller of the item you were viewing before you clicked advanced search. It can't find the route and WSOD with this in the logs.

The solution is to just remove the search context entirely, which I believe was a desired state anyway because we found it confusing.
